### PR TITLE
Remove sanitize copy requirement from the spec.

### LINKED ISSUE #150: Make async clipboard APIs (read/write) to sanitize interoperably with setData/getData for text/html
Hi All,
Excel online team reached out to us on several issues they found with clipboard formats. Sanitization across legacy and async clipboard APIs is not consistent for some mime types. For example, while text/html when set with setData/getData, keeps meta tags in Chrome, async clipboard APIs strip them down. 
This creates an issue for online text editing applications. Target apps, processing paste, rely on meta tags to infer information about the clipboard payload source. @snianu  has wrote a detailed analysis which can be found [here](https://docs.google.com/document/d/1rTEg2I-hMPXGiLrEMqKJz2Ycu6GRjlM3uvakOe84m8Q/edit#heading=h.yi2bvmwbg260). 
A matrix view of formats can be found [here](https://docs.google.com/spreadsheets/d/16GxpRzx77NaLTKsU5e1CZVtA5ehvxpZVFmJ-02A0uxs/edit?usp=sharing).
We propose that we make async clipboard read/write serialization behaviors consistent with legacy clipboard API such as setData/getData.
The reason being browsers already expose this information through setData/getData and making async clipboard read/write to behave in a compatible way which will ease its adoption over time. 

If we can all agree on this, we will follow up with a PR clarifying the behavior in the Clipboard Spec which will also take care of #140 

CC: @rniwa @whsieh  @dway123 @garykac @BoCupp-Microsoft @megangardner @johanneswilm

### PR DESCRIPTION
**Pull request by snianu**
Closes #150 

For normative changes, the following tasks have been completed:

 * [ ] Modified Web platform tests (link to pull request)

Implementation commitment:

 * [ ] WebKit (https://bugs.webkit.org/show_bug.cgi?id=)
 * [ ] Chromium (https://bugs.chromium.org/p/chromium/issues/detail?id=)
 * [ ] Gecko (https://bugzilla.mozilla.org/show_bug.cgi?id=)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/clipboard-apis/pull/174.html" title="Last updated on May 20, 2022, 6:50 PM UTC (467a4e6)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/clipboard-apis/174/14c72dc...467a4e6.html" title="Last updated on May 20, 2022, 6:50 PM UTC (467a4e6)">Diff</a>

### CODE REVIEW

**Comment by snianu** [comment]
@mbrodesser @annevk @BoCupp-Microsoft FYI. I added a note for Safari's sanitization behavior, but since FF and Chromium are interop and we have a desire to align async api with the DataTransfer API, I'm removing the sanitization language from the spec only for async apis. PTAL.

**Comment by annevk** [comment]
I don't think we should diverge on the internal details between these APIs. I don't really see what benefit that brings.

**Comment by snianu** [comment]
@annevk This change aligns async clipboard HTML read/write behavior with DataTransfer APIs for Chromium and Firefox browsers. Sanitization requirement in async clipboard APIs is a regression for us since popular apps like Office Online, Google docs, Adobe etc rely on clipboard APIs to not strip out content that it thinks is "harmful". Since Safari is the only browser that has a different behavior, any objection to making this change in the spec?

**Comment by annevk** [comment]
Yeah, I think this goes against what we discussed in the meetings.

**Comment by snianu** [comment]
In [this ](https://github.com/w3c/clipboard-apis/issues/150#issuecomment-1031684598)comment, you mentioned that you are in favor of aligning the behavior of DataTransfer and Async clipboard APIs. Am I misunderstanding something?

**Comment by annevk** [comment]
Yes and @whsieh and I argued for allowing sanitization at least for built-in types too. Those can both be true.

File: index.bs [hunk 1]
```diff
@@ -837,11 +837,11 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 					1. Abort these steps.
 
-				1. Let |data| be a copy of the [=system clipboard data=] represented as [=clipboard items=]. For the MIME types defined in the [=mandatory data types=] list, |data| contains the sanitized copy of text/plain format.
+				1. Let |data| be a copy of the [=system clipboard data=] represented as [=clipboard items=]. For the MIME types defined in the [=mandatory data types=] list, |data| contains the content of text/plain format.
```
Comment (BoCupp-Microsoft) [inline index.bs:840]: Suggest that we change:

>For the MIME types defined in the [=mandatory data types=] list, |data| contains the content of text/plain format.

to 

For the MIME types defined in the [=mandatory data types=] list, |data| MAY be sanitized.

File: index.bs [hunk 2]
```diff
@@ -837,11 +837,11 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
...
 
 					Issue: Some OSs contain multiple clipboard (e.g. Linux, "primary", "secondary", "selection"). Define from which of those data is read.
 
-					Issue: Add definition of sanitized copy.
+					Issue: Currently Safari does some sanitization. See https://github.com/w3c/clipboard-apis/issues/150.
```
Comment (BoCupp-Microsoft) [inline index.bs:844]: Not sure if we need this issue if we have agreement on saying data MAY be sanitized above.

File: index.bs [hunk 3]
```diff
@@ -953,17 +953,9 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 							1. If |type| is not in the [=mandatory data types=] list, then [=reject=] |p| with {{"NotAllowedError"}} {{DOMException}} in |realm| and abort these steps.
 
-							1. Let |cleanItem| be a sanitized copy of |blob|.
```
Comment (BoCupp-Microsoft) [inline index.bs:956]: Maybe instead change to be:

1. Let |cleanItem| be an optionally sanitized copy of |blob|

File: index.bs [hunk 4]
```diff
@@ -953,17 +953,9 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
...
-
-								Issue: Add definition of sanitized copy.
-
-							1. If unable to create a sanitized copy, then follow the below steps:
```
Comment (BoCupp-Microsoft) [inline index.bs:960]: Maybe change to be:

1. If sanitization was attempted and was not successfully completed, then follow the below steps:

File: index.bs [hunk 5]
```diff
@@ -1702,19 +1694,9 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 		1. Else
 
-			1. This is left to the implementation...
-
-				Issue: It's not good to leave things up to the
-				implementation. What should happen here?
+			1. Assign "" to |wellKnownFormat|.
```
Comment (BoCupp-Microsoft) [inline index.bs:1697]: This seems like it could be a separate PR as it isn't directly related to allowing an optional sanitization of the well-known formats?

File: index.bs [hunk 6]
```diff
@@ -786,15 +786,15 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 					1. Abort these steps.
 
-				1. Let |data| be a copy of the [=system clipboard data=] represented as [=clipboard items=]. For the MIME types defined in the [=mandatory data types=] list, |data| contains the sanitized copy of text/html format, but image/png format has unsanitized payload to preserve meta data.
+				1. Let |data| be a copy of the [=system clipboard data=] represented as [=clipboard items=]. For the MIME types defined in the [=mandatory data types=] list, |data| contains the content of text/html format, but image/png format has unsanitized payload to preserve meta data.
```
Comment (BoCupp-Microsoft) [inline index.bs:789]: Suggest that we change the sentence starting with:

> For the MIME types defined...

to 

For the MIME types defined in the [=mandatory data types=] list, |data| MAY be sanitized

File: index.bs [hunk 7]
```diff
@@ -786,15 +786,15 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
...
 
 					Issue: Some OSs contain multiple clipboard (e.g. Linux, "primary", "secondary", "selection"). Define from which of those data is read.
 
+					Issue: Currently Safari does some sanitization for [=mandatory data types=]. See https://github.com/w3c/clipboard-apis/issues/150.
```
Comment (BoCupp-Microsoft) [inline index.bs:793]: Same as I mentioned below... not sure if we need this called out if we are using language that makes it clear sanitization is optional.

**Comment by BoCupp-Microsoft** [comment]
> Yes and @whsieh and I argued for allowing sanitization at least for built-in types too. Those can both be true.

@annevk Yes we agree that sanitization can be allowed for built-in types.  I thought that we were in agreement also that it didn't need to be required.  Maybe the language I suggested in my review of this PR can help resolve the issue?

**Comment by annevk** [comment]
@BoCupp-Microsoft that's correct. Sanitization would be optional and the same for legacy/new APIs (ideally through them using a common algorithm).

File: index.bs (hunk 2 above)
Comment (snianu) [reply index.bs:844]: Removed.

File: index.bs (hunk 6 above)
Comment (snianu) [reply index.bs:789]: Done.

File: index.bs (hunk 7 above)
Comment (snianu) [reply index.bs:793]: Removed.

File: index.bs (hunk 1 above)
Comment (snianu) [reply index.bs:840]: Done.

File: index.bs (hunk 4 above)
Comment (snianu) [reply index.bs:960]: Done.

File: index.bs (hunk 5 above)
Comment (snianu) [reply index.bs:1697]: Done.

File: index.bs [hunk 8]
```diff
@@ -867,12 +867,16 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 										1. Resolve |p| with |v|.
 
+										1. Return |p|.
```
Comment (BoCupp-Microsoft) [inline index.bs:870]: Let's just make one change at a time focused on describing optional sanitization.  Could you remove these new returns and other editorial changes?

**Review by BoCupp-Microsoft — APPROVED** [review]
